simple_toml_settings/settings.py

```diff
@@ -71,8 +72,11 @@ def get_settings_folder(self) -> Path:
             return Path.home()
 
         settings_folder: Path = Path.home() / f".{self.app_name}"
+
+        if self.xdg_config:
+            settings_folder = Path.home() / f".config/{self.app_name}"
```

Comment:
Now that's an interesting point of philosophy:  I assume you're talking about AppData and ApplicationLibraries?  That's certainly not wrong, but xdg-aware (cli-mostly) programs save to my C:/Users/jhett/.config, not AppData and similar for Mac.  I use a  niche library [xdg-base-dirs](https://github.com/srstevenson/xdg-base-dirs) but for people developing desktop apps you're probably right.

Thank you for taking the time to review.

Reply:
And hard coding a path that can be altered due to environment variables as I'm sure that mature library you recommended does ... well, that just ain't right so I'll make the switch.

Reply:
Well .... 
You got me thinking so I checked into the package source. It **does** check the env variables but **only** under Linux (open contention between users/maintainers, with the latter refusing to budge). For Mac/Windows it just uses the default 'standards' for those systems (from Apple and MSN respectively).
As you say, this is cool under GUI apps but for cmdline apps that need manual editing of config files it is a pain. 

Since your whole PR is about getting **XDG compatibility**, i suggest we enforce that **regardless** of the OS.

So, how about instead of `platformdirs` we use the `xdg-base-dirs` you mentioned, which will enforce proper XDG standards on all without hard coding any folders while stil using env vars to override on any system? Apple and MS do not (and probably never will) standardize on this.

Later, I'll put another PR that offers **PLATFORM** specific folders as another option?

Sorry for the messing around!

Reply:
No worries.  While I've long been an xdg specc'r just because I suffer from analysis paralysis and having portable, predetermined buckets seems to finally be happening.  Your project will have to be lifted to 3.10 as a minimum to make that work though.  Are you ok with that?

Reply:
> Your project will have to be lifted to 3.10 as a minimum to make that work though. Are you ok with that?

Thats a pain. I did take a look through the code of that package and the only thing making it python 3.10 and not 3.9 is using `Path | None` as a return type on _one_ line instead of making it an `Optional[Path]`

for the sake of keeping 3.9 compatibilty (which still overall still has more users than 3.10) I'd be happier just checking for the relevant ENV variable and using if set, or `$HOME/.config` as default? Could just make a quick helper function that would do it.

Reply:
Sure.  That's the great thing about starting Python in earnest -- I can begin at the edge and be mainstream by the time I'm good!

Reply:
How are you getting on with this PR @borland502 , need any help?

Reply:
I'm afraid I've set it aside for the time being, having thought you were applying the finishing touches along with your own goals.  

I'm going on vacation tomorrow but can attend to it when I get back ofc, since I submitted the work.

Reply:
Sorry my bad, no I wasn't going to touch as it's your PR :grin:

I'm going to make a few underlying changes to the main class that will require some minor refactoring to your PR, but there is no rush for that. Enjoy your vacation, we can look at this when you get back and adjust if needed to fit any changes I've made.

Reply:
Well, it is a rainy day where I'm at so I made the change.  I had no idea that pipe was such a new feature.  I must say, I like it better than Union[]